pkg/controller/clusterconnection/clusterconnection_controller.go

```diff
@@ -207,6 +207,18 @@ func (r *ReconcileConnection) Reconcile(ctx context.Context, request reconcile.R
 		return reconcile.Result{}, err
 	}
 
+	preDefaultPatchFrom := client.MergeFrom(managementClusterConnection.DeepCopy())
+	fillDefaults(managementClusterConnection)
+
+	// Write the discovered configuration back to the API. This is essentially a poor-man's defaulting, and
+	// ensures that we don't surprise anyone by changing defaults in a future version of the operator.
+	// Note that we only write the 'base' installation back. We don't want to write the changes from 'overlay', as those should only
```

Comment:
> 	// Note that we only write the 'base' installation back. We don't want to write the changes from 'overlay', as those should only
> 	// be stored in the 'overlay' resource.

These lines only apply to the installation spec, not the management cluster connection.